### PR TITLE
Http mock

### PR DESCRIPTION
**Pull request by yoavtsook2806**

### CODE REVIEW

File: src/mockHttpModule.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import { isInstanceOfNetmockResponse } from './NetmockResponse';
+
+function handleNotMockedResponse(request: HttpRequest, requestCallback?: CallBack, isHttpsRequest?: boolean) {
+  const originalModule = isHttpsRequest ? global.originalHttps : global.originalHttp;
+  const func = originalModule.request;
+  const url = getUrlForHttp(request);
+  const method = getRequestMethodForHttp(request);
+  const initialResponseObject = {
+    headers: {},
+    location: 'BLA',
```
Comment (Niryo) [inline src/mockHttpModule.ts:16]: blalmos is better

File: src/mockHttpModule.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  };
+  if (isRealNetworkAllowed(url)) {
+    return func(request, requestCallback);
```
Comment (Niryo) [inline src/mockHttpModule.ts:19]: using `originalModule.request` directly instead saving it to func makes it more readable

File: src/utils.ts [hunk 3]
```diff
@@ -17,13 +39,23 @@ export function getMockedEndpointKey(input: RequestInfo | RegExp): string {
   return cleanUrl.endsWith('/') ? cleanUrl.slice(0, -1) : cleanUrl;
 }
 
+export function getMockedEndpointKeyForHttp(request: HttpRequest): string {
+  const parsedUrl = new URL(getUrlForHttp(request));
```
Comment (Niryo) [inline src/utils.ts:43]: don't we lose the ability to mock endpoint by regex like we do in the regular getMockedEndpointKey?

Comment (yoavtsook2806) [reply src/utils.ts:43]: Solved

**Review by Niryo — APPROVED** [review]

**Pull request merged**